thunder/executors/transformer_engineex.py

```diff
@@ -52,6 +53,7 @@
         TE_AVAILABLE = False
 
     TE_VERSION_1_3_PLUS = LooseVersion(version("transformer_engine")) >= LooseVersion("1.3")
+    TE_VERSION_1_6_PLUS = LooseVersion(version("transformer_engine")) > LooseVersion("1.6")
```

Comment:
Would transformer engine show up somewhere in requirements/ folder? Or is it an optional dependency?

Reply:
It's an optional dependency and all tests should be skipped if TransformerEngine is not installed. 
There's a PR that adds TransformerEngine to be tested in CI https://github.com/Lightning-AI/lightning-thunder/pull/209#issuecomment-2074407065